pkgs/pkgs-lib/formats.nix

```diff
@@ -473,4 +473,56 @@ rec {
     '') {};
   };
 
+  envvars = {
```

Comment:
These entries should have associated documentation in [`nixos/doc/manual/development/settings-options.section.md`](https://github.com/NixOS/nixpkgs/blob/44ea03ae51045e51387cbf6b5d6b35233fec0b56/nixos/doc/manual/development/settings-options.section.md#nix-representable-formats-json-yaml-toml-ini--sec-settings-nix-representable).

This should then be very explicit about what this format is and how it can be used:
- E.g. will it be usable by [systemd's `EnvironmentFile`](https://www.freedesktop.org/software/systemd/man/latest/systemd.exec.html#EnvironmentFile=)?
- I assume it will be `source`able in any POSIX shell (any reference to link to?)
- Which environment variables are allowed? Are all casings allowed? Does the case need to be normalised?

Reply:
I'll have a go at documenting it, addressing the questions. I'll ping you when it's ready, so you can look through it.